v21.1/null-handling.md

```diff
@@ -521,3 +521,33 @@ failed to satisfy CHECK constraint (price > 0)
 ~~~
 failed to satisfy CHECK constraint (discount <= price)
 ~~~
+
+## NULLs and concatenation with other types
+
+<span class="version-tag">New in v21.1</span> Concatenation between a non-`NULL` value and a `NULL` value results in a `NULL` value.
```

Comment:
I wonder if we should explicitly call out here that in v20.2 and earlier, that concatenation between a non-`NULL` value and a `NULL` value results in a `NULL` value would result in an array.

and that if you still need this behavior you can cast NULL to an array type.